apps/templates/SimpleAppAPI.h

```diff
@@ -19,4 +19,61 @@ int halide_buffer_display(const buffer_t* buffer);
 
 }  // extern "C"
 
+
+// This set of macros may be used to placing timing code around a halide
+// function to be benchmarked.
+#ifdef __APPLE__
+#define PROFILE_SHADER 1
+
+// In a program that calls the halide function repeatedly, e.g. once per frame
+// set the number of iterations to compute statistics over.
+#define PROFILE_ITERATIONS 1
+#ifdef PROFILE_SHADER
+#import <mach/mach_time.h>
+
+#define PROFILE_BEGIN(msg) \
+{\
+  uint64_t begin_time = mach_absolute_time();
```

Comment:
If we use halide_current_time_ns() here, we could dispense with the ifdef APPLE stuff